crowdsourcing/viewsets/task.py

```diff
@@ -259,3 +282,34 @@ class CurrencyViewSet(viewsets.ModelViewSet):
 
     queryset = Currency.objects.all()
     serializer_class = CurrencySerializer
+
+
+class ExternalSubmit(APIView):
+    def post(self, request, *args, **kwargs):
+        identifier = request.query_params.get('daemo_id', False)
+
+        if not identifier:
+            return Response("Missing identifier", status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            from django.conf import settings
+            from hashids import Hashids
+            hash = Hashids(salt=settings.SECRET_KEY)
+            task_worker_id, task_id, template_item_id = hash.decode(identifier)
+
+            with transaction.atomic():
+                task_worker = TaskWorker.objects.get(id=task_worker_id, task_id=task_id)
+                task_worker_result, created = TaskWorkerResult.objects.get_or_create(task_worker_id=task_worker.id,
+                                                                                     template_item_id=template_item_id)
+                # only accept in progress, submitted, or returned tasks
+                if task_worker.task_status in [1, 2, 5]:
+                    task_worker_result.status = 1
+                    task_worker_result.result = request.data
```

Comment:
I thought you already replied to that :)

Currently, multiple submits will overwrite the data if task is still in progress.

We expect maximum one form in an iframe but there is no technical limitation as such and we leave it to the author on how he wants to use it.

For a multi-step form, which can push data in partial mode, we can probably allow author to add another url parameter like - append - to signify data has to be appended to existing data. But we still expect component key names in form to be different in such case.

For all such extra options, we will need to instruct author somewhere on how these parameters can be used.